qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -1,5 +1,103 @@
+from functools import partial
+import numpy as np
+
 from qcodes import VisaInstrument
-from qcodes.utils.validators import Numbers, Ints, Enum, MultiType
+from qcodes.instrument.parameter import ArrayParameter
+from qcodes.utils.validators import Numbers, Ints, Enum, Strings
+
+
+class ChannelBuffer(ArrayParameter):
+    """
+    Parameter class for the two channel buffers
+
+    Currently always returns the entire buffer
+    TODO (WilliamHPNielsen): Make it possible to query parts of the buffer.
+        The instrument natively supports this in its TRCL call.
+    """
+
+    def __init__(self, name, instrument, channel):
+        """
+        Args:
+            name (str): The name of the parameter
+            instrument (SR830): The parent instrument
+            channel (int): The relevant channel (1 or 2). The name should
+                should match this.
+        """
+        if channel not in [1, 2]:
```

Comment:
I would make this an parameter and probably a tuple i.e. `self._valid_channels = (1,2)`